io/io/src/TFileMerger.cxx

```diff
@@ -591,8 +591,15 @@ Bool_t TFileMerger::MergeOne(TDirectory *target, TList *sourcelist, Int_t type,
       // GetPath(), so we can still figure out where we are in the recursion
 
       // If this folder is a onlyListed object, merge everything inside.
-      if (onlyListed) type &= ~kOnlyListed;
-      status = MergeRecursive(newdir, sourcelist, type);
+      const auto mergeType = onlyListed ? type &= ~kOnlyListed : type;
```

Comment:
This assignment inside a ternary operator is weird, please make them two separate lines

Reply:
I removed the assignment now (which could of caused a bug for subsequent directories), and extended the test to ensure that subsequent directories get the same treatment.